### PR TITLE
Update JDK versions

### PR DESCRIPTION
**Pull request by cushon**
* AdoptOpenJDK has moved to the Eclipse foundation and rebranded as Adoptium ([context](https://blog.adoptopenjdk.net/2021/03/transition-to-eclipse-an-update/)), so use Adoptium for `ppc64le` and `s390x`
* I am working on a script to use the Adoptium and Azul APIs for fetch the latest versions for the supported platforms (https://api.adoptium.net/q/swagger-ui/, https://app.swaggerhub.com/apis-docs/azul/zulu-download-community/1.0), and the buildozer API to update `distdir_deps.bzl`, which should make it easier to update these going forward especially now that the information is consolidated in `distdir_deps.bzl`.

### CODE REVIEW

**Comment by cushon** [comment]
@hvadehra @comius the new versions will need to be added to the Bazel mirror. What's the easiest way to do that? I can have the script I mentioned in the PR description provide a list of URLs in whatever format is most convenient

File: distdir_deps.bzl [hunk 1]
```diff
@@ -351,45 +351,49 @@ DIST_DEPS = {
...
             "openjdk11_linux_archive",
         ],
-        "archive": "zulu11.50.19-ca-jdk11.0.12-linux_x64.tar.gz",
-        "sha256": "b8e8a63b79bc312aa90f3558edbea59e71495ef1a9c340e38900dd28a1c579f3",
-        "strip_prefix": "zulu11.50.19-ca-jdk11.0.12-linux_x64",
+        "archive": "zulu11.52.13-ca-jdk11.0.13-linux_x64.tar.gz",
+        "sha256": "77a126669b26b3a89e0117b0f28cddfcd24fcd7699b2c1d35f921487148b9a9f",
+        "strip_prefix": "zulu11.52.13-ca-jdk11.0.13-linux_x64",
         "urls": [
-            "https://mirror.bazel.build/openjdk/azul-zulu11.50.19-ca-jdk11.0.12/zulu11.50.19-ca-jdk11.0.12-linux_x64.tar.gz",
```
Comment (cushon) [inline distdir_deps.bzl:358]: The location in the mirror wasn't consistent. I updated it to use the relative path after `https://cdn.azul.com/` from the original URL, but this is easy to change if you have a preference

File: distdir_deps.bzl [hunk 2]
```diff
@@ -351,45 +351,49 @@ DIST_DEPS = {
             "remotejdk11_linux_for_testing",
             "openjdk11_linux_archive",
         ],
-        "archive": "zulu11.50.19-ca-jdk11.0.12-linux_x64.tar.gz",
-        "sha256": "b8e8a63b79bc312aa90f3558edbea59e71495ef1a9c340e38900dd28a1c579f3",
-        "strip_prefix": "zulu11.50.19-ca-jdk11.0.12-linux_x64",
+        "archive": "zulu11.52.13-ca-jdk11.0.13-linux_x64.tar.gz",
```
Comment (cushon) [inline distdir_deps.bzl:354]: Some of these set `"archive"`, and some don't. Is this part actually needed?

**Comment by cushon** [comment]
I think the test failures are all because the new URLs aren't on the Bazel mirror yet

File: distdir_deps.bzl [hunk 3]
```diff
@@ -473,11 +483,11 @@ DIST_DEPS = {
             "remotejdk17_win_for_testing",
             "openjdk17_windows_archive",
         ],
-        "sha256": "f4437011239f3f0031c794bb91c02a6350bc941d4196bdd19c9f157b491815a3",
-        "strip_prefix": "zulu17.28.13-ca-jdk17.0.0-win_x64",
+        "sha256": "ab7c10f1cb76d10b801a4a130fee874e8e30e597fecb0afa9e76800e7d34aee7",
+        "strip_prefix": "zulu17.30.15-ca-jdk17.0.1-win_x64.zip",
```
Comment (PiotrSikora) [inline distdir_deps.bzl:487]: Drop `.zip`.

**Comment by PiotrSikora** [comment]
> I think the test failures are all because the new URLs aren't on the Bazel mirror yet

Some of those errors are because the `strip_prefix` doesn't match what's in the archive.

But there are also some build failures. You can click on `Details` to see the exact reason.

**Comment by cushon** [comment]
@PiotrSikora thanks! I had looked at the failures but obviously not carefully enough, the `strip_prefix` issue is fixed now.

The remaining failures all appear to be the expected ones:

```
test_verify_urls FAILED: Invalid urls: https://mirror.bazel.build/github.com/adoptium/temurin11-binaries/releases/download/jdk-11.0.13%2B8/OpenJDK11U-jdk_ppc64le_linux_hotspot_11.0.13_8.tar.gz https://mirror.bazel.build/github.com/adoptium/temurin11-binaries/releases/download/jdk-11.0.13%2B8/OpenJDK11U-jdk_s390x_linux_hotspot_11.0.13_8.tar.gz https://mirror.bazel.build/zulu-embedded/bin/zulu11.52.13-ca-jdk11.0.13-linux_aarch64.tar.gz https://mirror.bazel.build/zulu/bin/zulu11.52.13-ca-jdk11.0.13-linux_x64.tar.gz https://mirror.bazel.build/zulu/bin/zulu11.52.13-ca-jdk11.0.13-macosx_aarch64.tar.gz https://mirror.bazel.build/zulu/bin/zulu11.52.13-ca-jdk11.0.13-macosx_x64.tar.gz https://mirror.bazel.build/zulu/bin/zulu11.52.13-ca-jdk11.0.13-win_x64.zip https://mirror.bazel.build/zulu/bin/zulu17.30.15-ca-jdk17.0.1-linux_x64.tar.gz https://mirror.bazel.build/zulu/bin/zulu17.30.15-ca-jdk17.0.1-macosx_aarch64.tar.gz https://mirror.bazel.build/zulu/bin/zulu17.30.15-ca-jdk17.0.1-macosx_x64.tar.gz https://mirror.bazel.build/zulu/bin/zulu17.30.15-ca-jdk17.0.1-win_x64.zip.
/var/lib/buildkite-agent/.cache/bazel/_bazel_buildkite-agent/ec321eb2cc2d0f8f91b676b6d4c66c29/sandbox/linux-sandbox/6167/execroot/io_bazel/bazel-out/k8-fastbuild/bin/src/test/shell/bazel/verify_workspace.runfiles/io_bazel/src/test/shell/bazel/verify_workspace:62: in call to test_verify_urls
```

**Comment by cushon** [comment]
I tried removing all of the bazel mirrors and confirmed all of the tests pass.

@comius how do I get the new JDKs added to the mirror? I'm happy to do it if I can get the necessary permissions, otherwise I can provide the list of URLs and hashes in whatever format is most convenient

**Comment by comius** [comment]
> @comius how do I get the new JDKs added to the mirror? I'm happy to do it if I can get the necessary permissions, otherwise I can provide the list of URLs and hashes in whatever format is most convenient

Sorry for the delay, I'll upload them. Don't know who sets the permissions. @meteorcloudy perhaps, do you know?

> and the buildozer API to update distdir_deps.bzl, which should make it easier to update these going forward especially now that the information is consolidated in distdir_deps.bzl.

There is another effort - bzlmod - which can resolve Bazel dependecies and it will eventually outdate/remove distdir_deps.bzl. It's probably happening within a year or so. Something to consider, where to put our efforts.

File: distdir_deps.bzl (hunk 2 above)
Comment (comius) [reply distdir_deps.bzl:354]: If the dep is used in distdir_tar rule, then archive needs to be set.

File: distdir_deps.bzl [hunk 4]
```diff
@@ -351,45 +351,49 @@ DIST_DEPS = {
...
             "remotejdk11_linux_ppc64le_for_testing",
         ],
-        "sha256": "a417db0295b1f4b538ecbaf7c774f3a177fab9657a665940170936c0eca4e71a",
-        "strip_prefix": "jdk-11.0.7+10",
+        "sha256": "82f14cda71cff99c878bf8400598a87235adb6c81b0337f7077c27e5cac1190c",
+        "strip_prefix": "jdk-11.0.13+8",
         "urls": [
-            "https://mirror.bazel.build/openjdk/AdoptOpenJDK/openjdk11-binaries/releases/download/jdk-11.0.7+10/OpenJDK11U-jdk_ppc64le_linux_hotspot_11.0.7_10.tar.gz",
-            "https://github.com/AdoptOpenJDK/openjdk11-binaries/releases/download/jdk-11.0.7+10/OpenJDK11U-jdk_ppc64le_linux_hotspot_11.0.7_10.tar.gz",
+            "https://mirror.bazel.build/github.com/adoptium/temurin11-binaries/releases/download/jdk-11.0.13%2B8/OpenJDK11U-jdk_ppc64le_linux_hotspot_11.0.13_8.tar.gz",
```
Comment (meteorcloudy) [inline distdir_deps.bzl:383]: Does `%2B8` suppose to be here? I suspect this is breaking the verify_urls test: https://cs.opensource.google/bazel/bazel/+/master:src/test/shell/bazel/verify_workspace.sh;l=45?q=test_verify_urls

**Comment by meteorcloudy** [comment]
@cushon Can you give me a list of urls you want to mirror? But I don't think the test failure is caused by missing mirrors, see my other comment.

File: distdir_deps.bzl (hunk 4 above)
Comment (meteorcloudy) [reply distdir_deps.bzl:383]: Replacing `%2B` with `+` could probably work around the test failure, I also confirmed the url will still work.

**Review by comius — APPROVED** [review]
JDKs uploaded.

**Comment by cushon** [comment]
I fixed the url encoding issue, thanks.

The URLs are:

```
77a126669b26b3a89e0117b0f28cddfcd24fcd7699b2c1d35f921487148b9a9f https://cdn.azul.com/zulu/bin/zulu11.52.13-ca-jdk11.0.13-linux_x64.tar.gz
6be187379c26506a4b804b4f734c17e554aebe4204bde58a10b429054cc9cf9f https://cdn.azul.com/zulu-embedded/bin/zulu11.52.13-ca-jdk11.0.13-linux_aarch64.tar.gz
82f14cda71cff99c878bf8400598a87235adb6c81b0337f7077c27e5cac1190c https://github.com/adoptium/temurin11-binaries/releases/download/jdk-11.0.13+8/OpenJDK11U-jdk_ppc64le_linux_hotspot_11.0.13_8.tar.gz
9d280d86fdf6a7d9e5cbf54dc37f1d6d09dfe676ff5c684802fdfa3932eee63e https://github.com/adoptium/temurin11-binaries/releases/download/jdk-11.0.13+8/OpenJDK11U-jdk_s390x_linux_hotspot_11.0.13_8.tar.gz
e27a11a6e970ba6f597ecc957c0cdb502ff8990c243a6abd9df1e3413a0a3e44 https://cdn.azul.com/zulu/bin/zulu11.52.13-ca-jdk11.0.13-macosx_x64.tar.gz
eb8d593a61a9461a554e7bb1d67bca0f94242273f1d01a13f58c20c269b35fe5 https://cdn.azul.com/zulu/bin/zulu11.52.13-ca-jdk11.0.13-macosx_aarch64.tar.gz
49289cc181504a01c946d5a66a5c904f8e6d912a2226dd82db62eb37e6c0a9d9 https://cdn.azul.com/zulu/bin/zulu11.52.13-ca-jdk11.0.13-win_x64.zip
9b8e4d1e47b02b9c2392462ee82988c189357471de3224c37173fa58e2b25112 https://cdn.azul.com/zulu/bin/zulu17.30.15-ca-jdk17.0.1-linux_x64.tar.gz
09d64fe576373b4314422811bc8402fbb7700176822b0e1e2bf2ff8a6cad10eb https://cdn.azul.com/zulu/bin/zulu17.30.15-ca-jdk17.0.1-macosx_x64.tar.gz
ce10425ce9cefdfb23ebeabebc0944cfb41531114a2d5bd89e3c19cc5cfa9913 https://cdn.azul.com/zulu/bin/zulu17.30.15-ca-jdk17.0.1-macosx_aarch64.tar.gz
ab7c10f1cb76d10b801a4a130fee874e8e30e597fecb0afa9e76800e7d34aee7 https://cdn.azul.com/zulu/bin/zulu17.30.15-ca-jdk17.0.1-win_x64.zip
```

**Comment by cushon** [comment]
> JDKs uploaded.

Thanks! Sorry I was too slow to provide a list of URLs :)

**Comment by cushon** [comment]
> There is another effort - bzlmod - which can resolve Bazel dependecies and it will eventually outdate/remove distdir_deps.bzl. It's probably happening within a year or so. Something to consider, where to put our efforts.

Sounds great, thanks. I won't put much effort into 'productionizing' the script, but having a way to fetch the latest URls and hashes may still be useful with bzlmod

**Pull request closed without merging**